### PR TITLE
fix: KT-40021 `operator` modifier is not required when overriding operators

### PR DESCRIPTION
**Pull request by p7nov**
Fix for https://youtrack.jetbrains.com/issue/KT-40021

### CODE REVIEW

File: docs/topics/operator-overloading.md [hunk 1]
```diff
@@ -1,11 +1,26 @@
 [//]: # (title: Operator overloading)
 
-Kotlin allows us to provide implementations for a predefined set of operators on our types. These operators have fixed 
-symbolic representation (like `+` or `*`) and fixed precedence. To implement an operator, provide a [member function](functions.md#member-functions)
-or an [extension function](extensions.md) with a fixed name, for the corresponding type, that means a left-hand side type for binary operations and argument type for unary ones.
-Functions that overload operators need to be marked with the `operator` modifier.
+Kotlin allows you to provide custom implementations for the predefined set of operators on your types. These operators have
```
Comment (koshachy) [inline docs/topics/operator-overloading.md:3]: I have doubts about "your type". Does it necessary to specify exactly *whose* types?

File: docs/topics/operator-overloading.md [hunk 2]
```diff
@@ -1,11 +1,26 @@
...
+```
+When [overriding](inheritance.md#overriding-methods) your operator overloads, you can omit `operator`.
+
+```kotlin
+class OrdersList: IndexedContainer {
+    override fun get(index: Int) { /*...*/ }   
+}
+```
+
+Read on to learn the conventions that regulate operator overloading for different operators.
```
Comment (koshachy) [inline docs/topics/operator-overloading.md:23]: I don't know, do we need this line at all :)

File: docs/topics/operator-overloading.md [hunk 3]
```diff
@@ -1,11 +1,26 @@
...
-symbolic representation (like `+` or `*`) and fixed precedence. To implement an operator, provide a [member function](functions.md#member-functions)
-or an [extension function](extensions.md) with a fixed name, for the corresponding type, that means a left-hand side type for binary operations and argument type for unary ones.
-Functions that overload operators need to be marked with the `operator` modifier.
+Kotlin allows you to provide custom implementations for the predefined set of operators on your types. These operators have
+prefedined symbolic representation (like `+` or `*`) and precedence. To implement an operator, provide a [member function](functions.md#member-functions)
+or an [extension function](extensions.md) with a specific name for the corresponding type. This type becomes the left-hand side type
+for binary operations and the argument type for the unary ones.
 
-Further you describe the conventions that regulate operator overloading for different operators.
+To overload an operator, mark the corresponding function with the `operator` modifier.
```
Comment (koshachy) [inline docs/topics/operator-overloading.md:8]: Perhaps the colon sign is more appropriate here?

File: docs/topics/operator-overloading.md [hunk 4]
```diff
@@ -1,11 +1,26 @@
...
+
+```kotlin
+interface IndexedContainer {
+    operator fun get(index: Int)
+}
+```
+When [overriding](inheritance.md#overriding-methods) your operator overloads, you can omit `operator`.
```
Comment (koshachy) [inline docs/topics/operator-overloading.md:15]: Same as above.

**Comment by p7nov** [comment]
@koshachy thanks, agree on all issues :)

**Review by koshachy — APPROVED** [review]
👍

**Pull request merged**